lib/rules/unit-disallowed-list/index.js

```diff
@@ -96,75 +113,108 @@ const rule = (primary, secondaryOptions) => {
 			});
 		}
 
-		/**
-		 * @template {import('postcss').AtRule} T
-		 * @param {T} node
-		 * @param {string} value
-		 * @param {(node: T) => number} getIndex
-		 * @returns {void}
-		 */
-		function checkMedia(node, value, getIndex) {
-			mediaParser(node.params).walk(/^media-feature$/i, (mediaFeatureNode) => {
-				const mediaName = getMediaFeatureName(mediaFeatureNode);
-				const parentValue = mediaFeatureNode.parent.value;
-
-				valueParser(value).walk((valueNode) => {
-					// Ignore all non-word valueNode and
-					// the values not included in the parentValue string
-					if (valueNode.type !== 'word' || !parentValue.includes(valueNode.value)) {
-						return;
+		root.walkAtRules(/^media$/i, (atRule) => {
+			const params = getAtRuleParams(atRule);
+
+			if (!hasDimension(params)) return;
+
+			parseFromTokens(tokenizeWithoutPercentageTokens(params)).forEach((mediaQuery) => {
+				/** @type {{ mediaFeatureName?: string }} */
+				const state = {
+					mediaFeatureName: undefined,
+				};
+
+				mediaQuery.walk((entry) => {
+					if (!entry.state) return;
+
+					if (isMediaFeature(entry.node)) {
+						entry.state.mediaFeatureName = entry.node.getName().toLowerCase();
 					}
 
+					if (!isTokenNode(entry.node)) return;
+
 					check(
-						node,
-						getIndex(node),
-						valueNode,
-						mediaName,
-						secondaryOptions ? secondaryOptions.ignoreMediaFeatureNames : {},
+						atRule,
+						atRuleParamIndex,
+						entry.node,
+						entry.state?.mediaFeatureName,
```

Comment:
[ask] Does passing `mediaFeatureName` to `check()` make sense if `mediaFeatureName` is empty? Can we skip it like this?

```js
if (!entry.state.mediaFeatureName) return;

check(
  // ...
  entry.state.mediaFeatureName,
  // ...
);
```

Reply:
Yes we can skip it!
Good catch 🙇